packages/amplication-client/src/Components/EntitySelectField.tsx

```diff
@@ -11,27 +11,37 @@ type TEntities = {
   ];
 };
 
+type RelatedEntityOption = {
+  value: string;
+  label: string;
+};
+
 type Props = Omit<SelectFieldProps, "options"> & {
   resourceId: string;
 };
 
 const EntitySelectField = ({ resourceId, ...props }: Props) => {
+  const [relatedEntityListOptions, setRelatedEntityListOptions] =
+    React.useState<RelatedEntityOption[]>([]);
   const { data: entityList } = useQuery<TEntities>(GET_ENTITIES, {
     variables: {
       resourceId: resourceId,
     },
+    fetchPolicy: "no-cache",
   });
 
-  const entityListOptions = useMemo(() => {
-    return entityList
-      ? entityList.entities.map((entity) => ({
-          value: entity.id,
-          label: entity.displayName,
-        }))
-      : [];
+  useEffect(() => {
+    if (!entityList) return;
+
+    const relatedEntityOptions = entityList.entities.map((entity) => ({
+      value: entity.id,
+      label: entity.displayName,
+    }));
+
+    setRelatedEntityListOptions(relatedEntityOptions);
```

Comment:
why are we using useEffect + useState, instead of the useMemo that was used before?

Reply:
you are using useMemo in a wrong way. useMemo should memoized the result of a logic function. react component has a life cycle which controlled by state and state listeners (useEffect). this is the right way to do it, in sense of React tools and more important the render cycle .